File: go/rna-transcription/rna_transcription.go
// Package strand provides an RNA transcription method
package strand

// ToRNA returns the RNA complement for a given DNA strand.
func ToRNA(dna string) (rna string) {
	for _, v := range dna {
		switch v {
		case 'G':
			rna += "C"
		case 'C':
			rna += "G"
		case 'T':
			rna += "A"
		case 'A':
			rna += "U"
		}
	}

	return rna
}

File: typescript/rna-transcription/rna-transcription.ts
export const DnaRnaMap: Record<string, string> = {
  G: 'C',
  C: 'G',
  T: 'A',
  A: 'U',
}

class Transcriptor {
  toRna(dna: string): string {
    let rna = ''

    for (const el of dna) {
      const nucleotide = DnaRnaMap[el]
      if (!nucleotide) throw Error('Invalid input DNA.')
      rna += nucleotide
    }

    return rna
  }
}

export default Transcriptor
